core/src/test/java/org/elasticsearch/http/netty/HttpPublishPortIT.java

```diff
@@ -34,18 +35,37 @@
 import static org.hamcrest.Matchers.greaterThanOrEqualTo;
 import static org.hamcrest.Matchers.instanceOf;
 
-@ClusterScope(scope = Scope.SUITE, numDataNodes = 1)
+@ClusterScope(scope = Scope.SUITE, numDataNodes = 0)
 public class HttpPublishPortIT extends ESIntegTestCase {
-    @Override
-    protected Settings nodeSettings(int nodeOrdinal) {
-        return Settings.settingsBuilder()
-                .put(super.nodeSettings(nodeOrdinal))
-                .put(NetworkModule.HTTP_ENABLED.getKey(), true)
-                .put(HttpTransportSettings.SETTING_HTTP_PUBLISH_PORT.getKey(), 9080)
-                .build();
-    }
 
     public void testHttpPublishPort() throws Exception {
+        Settings settings = Settings.settingsBuilder()
```

Comment:
can we use package private methods and have unit tests for this.. an integration seems like an overkill.